packages/block-library/src/paragraph/edit.js

```diff
@@ -162,7 +164,7 @@ class ParagraphBlock extends Component {
 					className={ classnames( 'wp-block-paragraph', className, {
 						'has-text-color': textColor.color,
 						'has-background': backgroundColor.color,
-						'has-drop-cap': dropCap,
+						'has-drop-cap': showDropCap && dropCap,
```

Comment:
This change touches the editor, but do we want to accommodate that in the frontend too? It can be hard to know whether it's RTL or not in that case as it might change. Could this be just a CSS fix instead maybe?